NSS-Day-Cohort-58/rare-rest-quicksilver-dynamos
# Nm user profiles

## Tickets: 18/22
     

## Description

Created profile_view module; can now view all user profiles, individual profiles, toggle a user profile from admin to non-admin, and activate and de-activate profiles.

Added admin check to create in post.py; now a new post by an admin will be automatically approved and a post by a non-admin be unapproved

## Type of change


To test this branch:
  - Git fetch --all
  - Git checkout nm-user-profiles
  - [ ] Test A - Test in Postman by making a GET request to http://localhost:8000/profiles 
  - [ ] Test B - Test in Postman by making a GET request to http://localhost:8000/profiles/3
  - [ ] Test C - Test in Postman by making a PUT request to http://localhost:8000/profiles/3 with raw body: {"user": {
            "is_staff": true,
            "is_active": true
        }}
     then make a GET request to http://localhost:8000/3 to verify properties are now true.
- [ ] Test D - Test in Postman by using the admin Token 
        0be249c88238743e5b4a7ac370b5145730c28e20 
      in the Authorization headers; make a POST request to http://localhost:8000/posts with raw body:
{
        "author": 2,
        "category": 3,
        "title": "See this!",
        "publication_date": "2022-11-08",
        "image_url": "post_1_test_img",
        "content": "Ancient microchip.",
        "approved": false,
        "reactions": [],
        "tags": []
}
    make a GET request to http://localhost8000/posts to see approved was changed to true due to 
    using admin token.
- [ ] Test E - Test in Postman by using the non-admin Token 
        06787add96eb3264080015997720cb0877b40f90 in the Authorization headers; make a POST 
        request to http://localhost:8000/posts with raw body:
{
        "author": 2,
        "category": 3,
        "title": "See this!",
        "publication_date": "2022-11-08",
        "image_url": "post_1_test_img",
        "content": "Ancient microchip.",
        "approved": true,
        "reactions": [],
        "tags": []
}
    make a GET request to http://localhost8000/posts to see approved was changed to false due to 
    using non-admin token.

## Checklist:
